sdk/core/azure-core-http-okhttp/src/main/java/com/azure/core/http/okhttp/OkHttpAsyncHttpClient.java

```diff
@@ -162,7 +222,13 @@ private static class OkHttpCallback implements okhttp3.Callback {
         @SuppressWarnings("NullableProblems")
         @Override
         public void onFailure(okhttp3.Call call, IOException e) {
-            sink.error(e);
+            if (e.getMessage().startsWith("canceled due to") && e.getSuppressed().length == 1) {
```

Comment:
Do we want to rely on the error message string here? If the text in the exception changes in OkHttp, this condition will be obsolete. `sink.error(e)` works better as the whole exception will be propagated upwards and let the upper layers handle this error.